drivers/dma/dma_silabs_siwx917.c

```diff
@@ -0,0 +1,480 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "rsi_rom_udma_wrapper.h"
+#include "rsi_udma.h"
+#include "sl_status.h"
+
+#include <errno.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+
+#define DT_DRV_COMPAT                       silabs_siwx917_dma
+#define DMA_MAX_TRANSFER_COUNT              1024
+#define DMA_CH_PRIORITY_HIGH                1
+#define DMA_CH_PRIORITY_LOW                 0
+#define UDMA0_INSTANCE                      0
+#define ULP_UDMA_INSTANCE                   1
+#define VALID_BURST_LENGTH                  0
+#define UDMA_ADDR_INC_NONE                  0X03
+#define PERIPHERAL_REQUEST_DISABLE          0
+#define PERIPHERAL_REQUEST_ENABLE           1
+#define NEXT_BURST_DISABLE                  0
+#define SOURCE_PROTECT_CONTROL_DISABLE      0
+#define DESTINATION_PROTECT_CONTROL_DISABLE 0
+#define BURST_REQUEST_DISABLE               0
+
+LOG_MODULE_REGISTER(si91x_dma, CONFIG_DMA_LOG_LEVEL);
+
+struct dma_siwx917_config {
+	UDMA0_Type *reg;             /* UDMA register base address */
+	uint8_t channels;            /* UDMA channel count */
+	uint8_t irq_number;          /* IRQ number */
+	void (*irq_configure)(void); /* IRQ configure function */
+};
+
+struct dma_siwx917_data {
+	UDMA_Channel_Info *udma_channel_info;
+	dma_callback_t dma_callback; /* User callback */
+	void *cb_data;               /* User callback data */
+	uint32_t dma_rom_buff[30];   /* Buffers which stores UDMA handle*/
+				     /* related information */
```

Comment:
Why 120 bytes? Why using `uint32_t`?

Reply:
I initially referenced the dma_rom_buff declarations in UDMA.c. However, after reviewing the code, I discovered that the buffer size should match the size of the RSI_UDMA_DATACONTEXT_T structure, which is 24 bytes. I have updated the buffer size to 24 bytes and rerun the test cases.

Reply:
I initially referenced the dma_rom_buff declarations in UDMA.c. However, after reviewing the code, I discovered that the buffer size should match the size of the RSI_UDMA_DATACONTEXT_T structure, which is 24 bytes. I have updated the buffer size to 6 bytes and rerun the test cases.

Reply:
For curiosity, why `RSI_UDMA_DATACONTEXT_T dma_rom_buff` does not work?